File: tests/Generator/resource/rust/default/news.rs
use serde::{Serialize, Deserialize};
use meta::Meta;
use author::Author;

// An general news entry
#[derive(Serialize, Deserialize)]
pub struct News {
    #[serde(rename = "config")]
    config: Option<Meta>,

    #[serde(rename = "inlineConfig")]
    inline_config: Option<std::collections::HashMap<String, String>>,

    #[serde(rename = "mapTags")]
    map_tags: Option<std::collections::HashMap<String, String>>,

    #[serde(rename = "mapReceiver")]
    map_receiver: Option<std::collections::HashMap<String, Author>>,

    #[serde(rename = "tags")]
    tags: Option<Vec<String>>,

    #[serde(rename = "receiver")]
    receiver: Option<Vec<Author>>,

    #[serde(rename = "data")]
    data: Option<Vec<Vec<f64>>>,

    #[serde(rename = "read")]
    read: Option<bool>,

    #[serde(rename = "author")]
    author: Author,

    #[serde(rename = "meta")]
    meta: Option<Meta>,

    #[serde(rename = "sendDate")]
    send_date: Option<chrono::NaiveDate>,

    #[serde(rename = "readDate")]
    read_date: Option<chrono::NaiveDateTime>,

    #[serde(rename = "price")]
    price: Option<f64>,

    #[serde(rename = "rating")]
    rating: Option<u64>,

    #[serde(rename = "content")]
    content: String,

    #[serde(rename = "question")]
    question: Option<String>,

    #[serde(rename = "version")]
    version: Option<String>,

    #[serde(rename = "coffeeTime")]
    coffee_time: Option<chrono::NaiveTime>,

    #[serde(rename = "g-recaptcha-response")]
    captcha: Option<String>,

    #[serde(rename = "media.fields")]
    media_fields: Option<String>,

    #[serde(rename = "payload")]
    payload: Option<serde_json::Value>,

}

File: tests/Generator/resource/python/default/news.py
from pydantic import BaseModel, Field, GetCoreSchemaHandler, Tag
from pydantic_core import CoreSchema, core_schema
from typing import Any, Dict, Generic, List, Optional, TypeVar, Annotated, Union
import datetime
from .meta import Meta
from .author import Author


# An general news entry
class News(BaseModel):
    config: Optional[Meta] = Field(default=None, alias="config")
    inline_config: Optional[Dict[str, str]] = Field(default=None, alias="inlineConfig")
    map_tags: Optional[Dict[str, str]] = Field(default=None, alias="mapTags")
    map_receiver: Optional[Dict[str, Author]] = Field(default=None, alias="mapReceiver")
    tags: Optional[List[str]] = Field(default=None, alias="tags")
    receiver: Optional[List[Author]] = Field(default=None, alias="receiver")
    data: Optional[List[List[float]]] = Field(default=None, alias="data")
    read: Optional[bool] = Field(default=None, alias="read")
    author: Author = Field(alias="author")
    meta: Optional[Meta] = Field(default=None, alias="meta")
    send_date: Optional[datetime.date] = Field(default=None, alias="sendDate")
    read_date: Optional[datetime.datetime] = Field(default=None, alias="readDate")
    price: Optional[float] = Field(default=None, alias="price")
    rating: Optional[int] = Field(default=None, alias="rating")
    content: str = Field(alias="content")
    question: Optional[str] = Field(default=None, alias="question")
    version: Optional[str] = Field(default=None, alias="version")
    coffee_time: Optional[datetime.time] = Field(default=None, alias="coffeeTime")
    captcha: Optional[str] = Field(default=None, alias="g-recaptcha-response")
    media_fields: Optional[str] = Field(default=None, alias="media.fields")
    payload: Optional[Any] = Field(default=None, alias="payload")
    pass


